avm/res/compute/virtual-machine/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,331 @@
+  }
+}
+
+// Diagnostics
+// ===========
+module diagnosticDependencies '../../../../../../utilities/e2e-template-assets/templates/diagnostic.dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}01'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-${namePrefix}-evh-${serviceShort}'
+    eventHubNamespaceName: 'dep-${namePrefix}-evhns-${serviceShort}'
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}-${iteration}'
+  params: {
+    location: location
+    name: '${namePrefix}${serviceShort}'
+    computerName: '${namePrefix}winvm1'
+    adminUsername: 'VMAdmin'
+    imageReference: {
+      publisher: 'MicrosoftWindowsServer'
+      offer: 'WindowsServer'
+      sku: '2019-datacenter'
+      version: 'latest'
+    }
+    nicConfigurations: [
+      {
+        deleteOption: 'Delete'
+        ipConfigurations: [
+          {
+            applicationSecurityGroups: [
+              {
+                id: nestedDependencies.outputs.applicationSecurityGroupResourceId
+              }
+            ]
+            loadBalancerBackendAddressPools: [
+              {
+                id: nestedDependencies.outputs.loadBalancerBackendPoolResourceId
+              }
+            ]
+            name: 'ipconfig01'
+            pipConfiguration: {
+              publicIpNameSuffix: '-pip-01'
+              roleAssignments: [
+                {
+                  roleDefinitionIdOrName: 'Reader'
+                  principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+                  principalType: 'ServicePrincipal'
+                }
+              ]
+            }
+            zones: [
+              '1'
+              '2'
+              '3'
+            ]
+            subnetResourceId: nestedDependencies.outputs.subnetResourceId
+            diagnosticSettings: [
+              {
+                name: 'customSetting'
+                metricCategories: [
+                  {
+                    category: 'AllMetrics'
+                  }
+                ]
+                eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+                eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+                storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+                workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+              }
+            ]
+          }
+        ]
+        nicSuffix: '-nic-01'
+        roleAssignments: [
```

Comment:
RBAC is not required to be WAF-aligned. Please also make sure to remove any other property that may be redundant. 
This may also mean that (like in this case) the MSI may be redundant.

Reply:
done